### ISSUE TITLE
LaTeX: Support `siunitx` Commands

### ISSUE THREAD

**Issue by schrieveslaach** [jgm/pandoc#3587]
LaTeX provides the package `siunitx` to typeset quantities in a consistent way. Pandoc should provide basic support for this package. For example, following code should be translated:

```latex
\SI[round-precision=2]{1}{m} is equal to \SI{1000}{mm}.

\SI[round-precision=2]{1}[\$]{} is equal to \SI{0.938094}{\euro}
```

Pandoc should parse this input example into following AST:

```haskell
[Para [Str "1m",Space,Str "is",Space,Str "equal",Space,Str "to",Space,Str "1000mm"]
,Para [Str "$1",Space,Str "is",Space,Str "equal",Space,Str "to",Space,Str "0.938094\8364"]]
```

**Comment by njbart** [comment]
Please note that “1m”, “1000mm” etc. are incorrect. According to http://www.bipm.org/en/publications/si-brochure/section5-3-3.html, “a space is always used to separate the unit from the number”.

(“The only exceptions to this rule are for the unit symbols for degree, minute, and second for plane angle, °, ', and ", respectively, for which no space is left between the numerical value and the unit symbol.”)

**Comment by schrieveslaach** [comment]
@njbart, done (except for degree, …). See PR #3588

**Comment by rriemann** [comment]
Can you please reopen this issue?

The siunitx command `\num{1000}` is missing that takes care for typesetting of numbers without unit. There is also a variant `\num[options]{1000}`.

The command will i.e. introduce a thin space or comma as a thousand marker, or convert the number notation to scientific or whatnot.

**Comment by jgm** [comment]
@schrieveslaach can you add support for `\num` too?

**Comment by schrieveslaach** [comment]
@jgm, yes I can. I would like to support expressions like `\num{1+-2i}`. However, I'm not sure what is the best approach to parse it correctly. Should we provide a math environment here? Can you provide any pointers how I should solve this?

**Comment by jgm** [comment]
I've not used the package, but from a brief glance, it looks
like they have their own ad hoc parsing of complex numbers,
exponentiation, and multiplication.  This could be
supported by custom parsers -- it's a fair bit of effort,
though, and could add a lot of bulk to the reader.

+++ schrieveslaach [Feb 07 18 08:37 ]:

**Comment by schrieveslaach** [comment]
@jgm, I'm aware that this might be complex and that I need to write some complex parsers, however, I was wondering if the parsers should just parse, for example, exponential expressions as UTF8 string or as some mathematical pandoc type. Do you have any recommendations? This could improve #4319 too.

**Comment by why-fi** [comment]
Since SI translations can be quite tedious, e.g.
`\SI[separate-uncertainty = true]{5(2)e3}{\meter}`
would result into
`$(5 \pm 2) \cdot 10^{3}~\text{m}`
I would recommend to use some mathematical pandoc type, if you are still working on the issue.

**Comment by rriemann** [comment]
Is the simple `\num` without option and complex numbers supported in the current master?

**Comment by njbart** [comment]
> I think between numbers and units should be a non-breakable half space instead of a normal space.

Non-breakable might make sense, but according to the BIPM website (see [my comment above](https://github.com/jgm/pandoc/issues/3587#issuecomment-294848036)), “a space is always used to separate the unit from the number”. – “Thin space” is mentioned only once on their whole website (and in a different context: “for numbers with many digits the digits may be divided into groups of three by a thin space“), and “half space” doesn’t occur at all.

**Comment by mb21** [comment]
@njbart From https://english.stackexchange.com/a/2911/89285:

> Note that the rule doesn't specify how wide the space should be. Some references recommend a normal inter-word space, while others recommend a thin space. In any case, the space is nonbreakable.

Ultimately, I guess this up to personal taste.

**Comment by tarleb** [comment]
It seems that this became a duplicate of #6658, which has been resolved.
